main.cpp

```diff
@@ -1859,9 +1861,18 @@ bool load_command::execute(device_map &devices) {
                     file_buf.insert(file_buf.begin(), read_range.from - aligned_range.from, 0);
                     file_buf.insert(file_buf.end(), aligned_range.to - read_range.to, 0);
                     assert(file_buf.size() == FLASH_SECTOR_ERASE_SIZE);
+
                     con.exit_xip();
-                    con.flash_erase(aligned_range.from, FLASH_SECTOR_ERASE_SIZE);
-                    raw_access.write_vector(aligned_range.from, file_buf);
+                    bool skip = false;
+                    if (settings.load.update) {
+                      vector<uint8_t> read_device_buf;
```

Comment:
Would it be better to declare this vector outside of the `for` loop? (i.e. just below `vector<uint8_t> device_buf;`)

Reply:
nah